src/Dialect/ONNX/ONNXOps/ShapeHelper.cpp

```diff
@@ -384,6 +384,33 @@ void updateType(Value val, ArrayRef<int64_t> shape, Type elementType,
   val.setType(resType);
 }
 
+static void resetTypeShapeToQuestionmarks(Value val) {
+  // Only deal with ranked tensor types here.
+  RankedTensorType valType = val.getType().dyn_cast<RankedTensorType>();
+  if (!valType)
+    return;
+  // Get info from ranked type.
+  ArrayRef<int64_t> shape = valType.getShape();
+  Type elementType = getElementType(valType);
+  Attribute encoding = valType.getEncoding();
+  // Reset any compile time literal to unknown (aka question marks).
+  SmallVector<int64_t, 4> newShape(shape.size(), -1);
+  // Build result type.
+  RankedTensorType resType;
+  if (encoding)
+    resType = RankedTensorType::get(newShape, elementType, encoding);
+  else
+    resType = RankedTensorType::get(newShape, elementType);
+  // Reset type
+  val.setType(resType);
+}
```

Comment:
I wonder if we can write this function in a more concrete way, say
```C++
// Only deal with ranked tensor types here.
RankedTensorType valType = val.getType().dyn_cast<RankedTensorType>();
if (!valType)
    return;
// Reset any compile time literal to unknown (aka question marks).
SmallVector<int64_t, 4> newShape(valType.getRank(), -1);
auto resType = RankedTensorType::Builder(valType).setShape(newShape);
// Reset type
val.setType(resType);
```

I haven`t tested the code, but hope it works.

Reply:
totally make sense.